### PR TITLE
SAK-41309: Error when linking an assignment with GradebookNG

### PR DESCRIPTION
**Pull request by jesusmmp**
In the table GB_GRADABLE_OBJECT_T always appear the name of the Gradebook item,
so I've changed the name of the function from "getAssignmentScoreStringByNameOrId"
to "getAssignmentScoreStringByName".

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -5059,7 +5059,7 @@ else if ("remove".equals(addUpdateRemoveAssignment)) {
                                                     (gradeStringToUse != null && aSubmission.getGradeReleased()) ? gradeStringToUse : "");
                                             gradebookExternalAssessmentService.updateExternalAssessmentComment(gradebookUid, associateGradebookAssignment, submitter.getSubmitter(),
                                                     (commentString != null && aSubmission.getGradeReleased()) ? commentString : "");
-                                        } else if (gradebookService.isAssignmentDefined(gradebookUid, associateGradebookAssignment)) {
```
Comment (jesusmmp) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Defined on line: 4948. Not need a new call

**Comment by steveswinsburg** [comment]
We have discussed previously about this being really fragile, since if the assignment name changes the link breaks. Best to use an ID. From working in this area, did you see how easy it might be to make this change? Would need to be a corresponding change on the Assignment side though.

**Comment by jesusmmp** [comment]
Hi @steveswinsburg, now we can't work in that solution. 
My patch resolve this: https://github.com/sakaiproject/sakai/pull/6296/files
If dont apply this patch then I think would reverse SAK-40650.

**Comment by ern** [comment]
see #6544 for another way to solve this issue.

**Pull request closed without merging**